frontend/src/views/Game.tsx

```diff
@@ -10,20 +16,45 @@ type LocationState = {
 function GamePage() {
   const location = useLocation<LocationState>();
   const [room, setRoom] = useState<Room | null>(null);
+  const [problems, setProblems] = useState<Problem[] | null>(null);
+  const [error, setError] = useState<string | null>(null);
 
   // Called every time location changes
   useEffect(() => {
     if (location && location.state && location.state.room) {
       setRoom(location.state.room);
     }
+    getProblems().then((res) => {
+      if (isError(res)) {
+        setError((res as ErrorResponse).message);
+        setProblems(null);
```

Comment:
I believe this is superfluous, as it is already set above.

Reply:
Resetting to null is safety for when this `useEffect()` function is called multiple times in one session. Originally, my `componentsDidMount()` equivalent would only have been called once but after merging with Alan's code it's supposed to rerun with every location change.